### PR TITLE
Added Steam and multithread support

### PR DESCRIPTION
**Pull request by claytonmgravatt**

### CODE REVIEW

File: maindata.py [hunk 1]
```diff
@@ -4,32 +4,44 @@
...
 
 #input string
 q = 'gravity'
 
-#adapt string for reddit or imdb query
-queryr = q.replace(" ","+")
-queryi = q.replace(" ","-")
-queryw = q.replace(" ","+")
-queryy = q.replace(" ","+")
+#tic = time.perf_counter()
```
Comment (gregv) [inline maindata.py:24]: Love that you are measuring performance and using threading for optimizations! 🥳

File: maindata.py [hunk 2]
```diff
@@ -4,32 +4,44 @@
...
+#Run reddit in seperate thread to reduce execution time
+rThread = pool.apply_async(redditFunction.getPushshiftData, (100, 1526428800,1589587200, q))
+
+i = imdbFunction.getIMDB(q)
+w = wikipediaFunction.getWiki(10, q)
+y = youtubeFunction.getYouTube(q)
+s = steamFunction.getSteam(q)
+
+#get thread running for reddit's output
+r = rThread.get()
```
Comment (gregv) [inline maindata.py:40]: Not sure it's easier, but Python 3.6 added concurrent > futures which makes it pretty easy to thread things - https://docs.python.org/3/library/concurrent.futures.html - just an FYI

File: redditfunction.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
 class RedditFunction :
     
     #This function returns a list of reddit post links
-    def getPushshiftData(self, size, after, before, queryr) :
+    def getPushshiftData(self, size, after, before, q) :
```
Comment (gregv) [inline redditfunction.py]: Try to look for libraries that make things easier - in this case, PRAW already exists as a Reddit wrapper https://praw.readthedocs.io/en/latest/ - if you can leverage modules that already exist, it saves you debugging time and makes your code more simple

File: youtubefunction.py [hunk 4]
```diff
@@ -6,8 +6,9 @@
 
 class YoutubeFunction :
     #This function returns a list of youtube video URLs
-    def getYouTube(self, queryy) :
+    def getYouTube(self, q) :
         
+            queryy = q.replace(" ","+")
             url = ('https://www.youtube.com/results?search_query=' +str(queryy) + '&sp=CAMSAhAB')
```
Comment (gregv) [inline youtubefunction.py:14]: Just saw `&sp=CAMSAhAB` - can you add a comment that talks about why you have this extra URL field?

File: youtubefunction.py [hunk 5]
```diff
@@ -18,11 +19,12 @@ def getYouTube(self, queryy) :
             
             ytLinkList = []
             
+            #find links to videos
             for link in soup.find_all('a', href= re.compile('watch')) :
                 if('https://www.youtube.com/' + link['href'] not in ytLinkList) :
                     ytLinkList.append('https://www.youtube.com/' + link['href'])
                     
-            return ytLinkList
+            return ytLinkList[:10]
```
Comment (gregv) [inline youtubefunction.py]: Why only the first 10? Can you add a comment that talks about your choice to only get the top 10

**Review by gregv — CHANGES_REQUESTED** [review]

**Comment by gregv** [comment]
@claytonmgravatt - a couple asks - marked this PR as "request changes"

File: redditfunction.py (hunk 3 above)
Comment (claytonmgravatt) [reply redditfunction.py:8]: I'm getting an error when I try to import praw: 
`ModuleNotFoundError: No module named 'praw'`
I've tried reinstalling, `pip3 install praw --upgrade --force-reinstall`, and I can see the package is there in the site-packages directory in file explorer, but I can't import it in python.

File: youtubefunction.py (hunk 5 above)
Comment (claytonmgravatt) [reply youtubefunction.py]: I've added a size parameter that's adjusted in maindata.py file. I had originally set it to 10 for consistency between the functions, but I've changed that to a variable number.

File: maindata.py (hunk 2 above)
Comment (claytonmgravatt) [reply maindata.py:40]: I implemented this (see below code snippet) and the output appears to be much slower than using the multiprocessing.pool _(on average it takes roughly 5 seconds more, which is still an improvement from no multi threading support)_ 

I was pretty surprised by this, so did a quick google search and found [this on stack overflow](https://stackoverflow.com/questions/18671528/processpoolexecutor-from-concurrent-futures-way-slower-than-multiprocessing-pool) 

``` python
# -*- coding: utf-8 -*-
from redditfunction import RedditFunction
from imdbfunction import ImdbFunction
from csvoutput import CsvOutput
from wikipediafunction import WikipediaFunction
from youtubefunction import YoutubeFunction
from steamfunction import SteamFunction
import time
from concurrent.futures import ThreadPoolExecutor

redditFunction = RedditFunction()
imdbFunction = ImdbFunction()
csvOutput = CsvOutput()
wikipediaFunction = WikipediaFunction()
youtubeFunction = YoutubeFunction()
steamFunction = SteamFunction()

#input string
q = 'gravity'
size = 10
tic = time.perf_counter()

#Run reddit in seperate thread to reduce execution time
with ThreadPoolExecutor(max_workers=1) as executor:
    rThread = executor.submit(redditFunction.getPushshiftData, 100, 1526428800,1589587200, q, size)
    
i = imdbFunction.getIMDB(q, size)
w = wikipediaFunction.getWiki(q, size)
y = youtubeFunction.getYouTube(q, size)
s = steamFunction.getSteam(q, size)

#get thread running for reddit's output
r = rThread.result()

toc = time.perf_counter()
print(f"did the thing in {toc - tic:0.4f} seconds")

#URL lists
print(r)
print(i)
print(w)
print(y)
print(s)
mylist = r + i + w + y + s

csvOutput.csvwrite(mylist, q)
```

**Pull request merged**